collect_app/src/main/java/org/odk/collect/android/database/DatabaseInstancesRepository.java

```diff
@@ -2,104 +2,148 @@
+import org.odk.collect.android.injection.DaggerUtils;
 import org.odk.collect.android.instances.Instance;
 import org.odk.collect.android.instances.InstancesRepository;
-import org.odk.collect.android.provider.InstanceProviderAPI.InstanceColumns;
 import org.odk.collect.android.storage.StoragePathProvider;
 
+import java.io.File;
+import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
 
+import static android.provider.BaseColumns._ID;
+import static org.odk.collect.android.database.DatabaseConstants.INSTANCES_TABLE_NAME;
+import static org.odk.collect.android.provider.InstanceProviderAPI.InstanceColumns.CAN_EDIT_WHEN_COMPLETE;
 import static org.odk.collect.android.provider.InstanceProviderAPI.InstanceColumns.DELETED_DATE;
+import static org.odk.collect.android.provider.InstanceProviderAPI.InstanceColumns.DISPLAY_NAME;
+import static org.odk.collect.android.provider.InstanceProviderAPI.InstanceColumns.GEOMETRY;
+import static org.odk.collect.android.provider.InstanceProviderAPI.InstanceColumns.GEOMETRY_TYPE;
+import static org.odk.collect.android.provider.InstanceProviderAPI.InstanceColumns.INSTANCE_FILE_PATH;
 import static org.odk.collect.android.provider.InstanceProviderAPI.InstanceColumns.JR_FORM_ID;
 import static org.odk.collect.android.provider.InstanceProviderAPI.InstanceColumns.JR_VERSION;
+import static org.odk.collect.android.provider.InstanceProviderAPI.InstanceColumns.LAST_STATUS_CHANGE_DATE;
+import static org.odk.collect.android.provider.InstanceProviderAPI.InstanceColumns.STATUS;
+import static org.odk.collect.android.provider.InstanceProviderAPI.InstanceColumns.SUBMISSION_URI;
+import static org.odk.collect.android.utilities.InstanceUtils.getInstanceFromCurrentCursorPosition;
+import static org.odk.collect.android.utilities.InstanceUtils.getValuesFromInstance;
 
 /**
  * Mediates between {@link Instance} objects and the underlying SQLite database that stores them.
- * <p>
- * Uses {@link InstancesDao} to perform database queries. {@link InstancesDao} provides a thin
- * convenience layer over {@link org.odk.collect.android.provider.InstanceProvider} which exposes
- * {@link Cursor} and {@link androidx.loader.content.CursorLoader} objects that need to be managed.
- * This can be advantageous when providing data to Android components (e.g. lists through adapters)
- * but is cumbersome in domain code and makes writing test implementations harder.
- * <p>
- * Over time, we should consider redefining the responsibility split between
- * {@link org.odk.collect.android.provider.InstanceProvider}, {@link InstancesRepository} and
- * {@link InstancesDao}.
  */
 public final class DatabaseInstancesRepository implements InstancesRepository {
 
+    private final InstancesDatabaseProvider instancesDatabaseProvider;
+
+    public DatabaseInstancesRepository() {
+        instancesDatabaseProvider = DaggerUtils.getComponent(Collect.getInstance()).instancesDatabaseProvider();
```

Comment:
Why don't you pass it as a parameter like we do in probably all the other cases like that?

Reply:
Yeah agreed this isn't great. I'm just cleaning this up in a follow up PR - I'm removing the zero arg constructors in the repo implementation. I just had them in to make it easier to construct the repos while concentrating on swapping out the cursor/content provider stuff.